cmd/rpcdaemon/commands/engine_api.go

```diff
@@ -2,41 +2,35 @@ package commands
 
 import (
 	"context"
+	"math/big"
 
+	"github.com/holiman/uint256"
+	"github.com/ledgerwatch/erigon-lib/gointerfaces"
+	types2 "github.com/ledgerwatch/erigon-lib/gointerfaces/types"
 	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/cmd/rpcdaemon/services"
 	"github.com/ledgerwatch/erigon/common"
 	"github.com/ledgerwatch/erigon/common/hexutil"
-	"github.com/ledgerwatch/erigon/rpc"
-)
-
-// The following code follows the Proof-of-stake specifications: https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.2/src/engine/interop/specification.md
-
-// Status codes returned by the engine_ API
-// - "SUCCESS": Operation ended successfully
-// - "INVALID": Block could not be execute properly (Bad block)
-// - "SYNCING": Execution layer is still syncing and trying to catch up to the block head.
-const (
-	engineSuccessCode = "SUCCESS"
-	//	engineInvalidCode = "INVALID"
-	engineSyncingCode = "SYNCING"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/log/v3"
 )
 
 // ExecutionPayload represents an execution payload (aka slot/block)
 type ExecutionPayload struct {
-	ParentHash    *rpc.BlockNumberOrHash `json:"parentHash"`
-	StateRoot     *rpc.BlockNumberOrHash `json:"stateRoot"`
-	ReceiptRoot   *rpc.BlockNumberOrHash `json:"receiptRoot"`
-	LogsBloom     *hexutil.Bytes         `json:"logsBloom"`
-	Random        *rpc.BlockNumberOrHash `json:"random"`
-	BlockNumber   *hexutil.Uint64        `json:"blockNumber"`
-	GasLimit      *hexutil.Uint64        `json:"gasLimit"`
-	GasUsed       *hexutil.Uint64        `json:"gasUsed"`
-	Extra         *hexutil.Bytes         `json:"extraData"`
-	BaseFeePerGas *hexutil.Big           `json:"baseFeePerGas"`
-	BlockHash     *rpc.BlockNumberOrHash `json:"blockHash"`
-	Timestamp     *hexutil.Uint64        `json:"timestamp"`
-	Coinbase      *common.Address        `json:"coinbase"`
-	Transactions  *[]hexutil.Bytes       `json:"transactions"`
+	ParentHash    common.Hash     `json:"parentHash"`
+	StateRoot     common.Hash     `json:"stateRoot"`
+	ReceiptRoot   common.Hash     `json:"receiptRoot"`
+	LogsBloom     hexutil.Bytes   `json:"logsBloom"`
+	Random        common.Hash     `json:"random"`
+	BlockNumber   hexutil.Uint64  `json:"blockNumber"`
+	GasLimit      hexutil.Uint64  `json:"gasLimit"`
+	GasUsed       hexutil.Uint64  `json:"gasUsed"`
+	Extra         hexutil.Bytes   `json:"extraData"`
+	BaseFeePerGas *hexutil.Big    `json:"baseFeePerGas"`
+	BlockHash     common.Hash     `json:"blockHash"`
+	Timestamp     hexutil.Uint64  `json:"timestamp"`
+	Coinbase      common.Address  `json:"coinbase"`
```

Comment:
Should be called "feeRecipient" instead per the [Engine API](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#executionpayloadv1).

Reply:
Ah, actually no, sorry, it's "coinbase" in [v1.0.0-alpha.4](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.4/src/engine/specification.md#executionpayloadv1).